transbank/webpay/webpayplus/transaction.py

```diff
@@ -56,3 +58,37 @@ def create(cls, buy_order, session_id, amount,
             else:
                 raise TransactionCreateException(http_response.status_code, e.args)
 
+    @classmethod
+    def commit(cls, token_ws, options=None):
+        commerce_code = None
+        api_key = None
+        base_url = None
+        if options is None:
+            commerce_code = WebpayPlus.commerce_code()
+            api_key = WebpayPlus.api_key()
+            base_url = WebpayPlus.integration_type_url()
+        else:
+            commerce_code = options.commerce_code
+            api_key = options.api_key
+            WebpayPlus.integration_type_url = options.integration_type
+            base_url = WebpayPlus.integration_type_url()
+
+        headers = {
+            "Tbk-Api-Key-Id": commerce_code,
+            "Tbk-Api-Key-Secret": api_key,
+            "Content-Type": "application/json",
+        }
+
+        http_client = WebpayPlus.http_client
+        final_url = base_url + cls.COMMIT_TRANSACTION_ENDPOINT + "/" + token_ws
+
+        try:
+            http_response = http_client.put(final_url, headers=headers)
+            http_response.raise_for_status()
+            response_json = http_response.json()
+            return TransactionCommitResponse(response_json)
+        except Exception as e:
+            if 'http_response' in locals():
```

Comment:
This is a code smell. Couple of questionable things happenning here:

1. The same code could be simpler if you just set `http_response` to None before the `try` statement.
2. If you want to be sure that `http_response` is available, why not use a context manager instead? (like `with http_client.put(final_url, headers=headers) as http_response:`) Isn't it supported by the requests library?
3. We are swallowing exception details. If anything useful came in the original `e` exception, we are losing all that information and putting a weird `-1` in there.  If we are supporting only python3 and up, you can use exception chaining here to avoid the problem. If not, I'd suggest you only catch some expected exceptions which you know you can handle without losing information — like a http 4xx o 5xx — and just let other exceptions bubble up. 

Yet again I suggest you look at the onepay code as a guide on how to deal with this things.

Reply:
1. Esta mañana plantee dos soluciones a eso, la primera de esas soluciones que propones @leosoto y la segunda es la de verificar si la variable está declarada en el conjunto de variables locales en la instancia de esa clase. Admito que no tomé la primera solución, para evitarme algún comentario que me diga que tengo una línea de código de más y qué pudo haberse manejado con la solución que implementé en este caso.

2. Si la respuesta de la librería `requests` es fallida porque sencillamente en un caso extremo, ese servidor se desconectó de la red, python lanza una excepción donde indica que el nombre de la variable (en este caso `http_response`) no se encuentra declarado, así tal cual hice literalmente la prueba , desconectando el computador de la red  y poder comprobarlo.

3. Ayer escribí el código para ir desde una excepción específica hasta una general, lo cual consistió en capturar 4 tipos de excepciones distintas, justamente para evitar que se utilice la clase muy general (`Exception`), eso lo puedes verificar directamente en commits anteriores (). Efectivamente también había un tipo de excepción que tomaba los errores http 4xx o 5xx, a pesar de ello hay un manejo de excepciones en caso de que sea válida la petición http pero no satisfactoria para el integrador (es decir aunque el servidor le responda pero no obtenga un código 2xx) pueda generar la excepción y pasar tanto código http como descripción del error al integrador, por lo tanto quedar manejada la excepción correctamente.

Reply:
Don't do things to "avoid comments". This kind of discussion exists to make better code and better products. If you think you have a point, stick to it and show people why you are right (while being open to critique at the same time: strong opinions, weakly held).

Back to the 3 points:

1. One extra line of code is a fair price for more readability. 
2. I understand all of that. You are not addressing my comment. I'm talking about using a `with` statement to simplify the code and avoid all that complication.
3. I don't think you are addressing the problem here either (swallowing exceptions and losing information)

Reply:
@leosoto 
1. The line with new var was added
2. I've tried with `with`statement with unsatified results
3. I've caught the exception, take the Exception message and the custom message from Transbank too, incorporate at the time of show the exception message